### PR TITLE
refactor Dependency.js to es6

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**

current tests should pass

**If relevant, link to documentation update:**

N/A

**Summary**

upgrade Dependency.js to es6

**Does this PR introduce a breaking change?**

No

**Other**
Dont expose compareLocations as Dependency attribute anymore as its only used internally anyways

### CODE REVIEW

**Comment by timse** [comment]
should the file be moved to `/dependencies` ?

**Comment by TheLarkInn** [comment]
@timse so this one is a little more involved to switch over because it is a top level superclass there are multiple levels of inheritance between this one and any other file that ends in Dependency. 

So to refactor this one, you would also need to first refactor any file that ends in Dependency. Good example is Dependency => NullDependency => ConstDependency. If your not quite up for that much refactoring that is okay, feel free to close this PR and work on the lower level (in the hierarchy) Dependency subclasses like anything in the dependency folder. Let me know if any of this is unclear.

**Comment by timse** [comment]
i guess i need to start at the top of the foodchain here to make this work :)

File: lib/Dependency.js [hunk 1]
```diff
@@ -2,43 +2,45 @@
...
+	}
 
-Dependency.compare = function(a, b) {
-	return Dependency.compareLocations(a.loc, b.loc);
-};
+	disconnect() {
+		this.module = null;
+	}
 
-Dependency.compareLocations = require("./compareLocations");
```
Comment (Kovensky) [inline lib/Dependency.js]: This is technically an API change; nothing left around that assumes it can find `compareLocations` in the constructor?

Comment (timse) [inline lib/Dependency.js]: nope :)

**Comment by timse** [comment]
now dependes on #3812 and #3818

**Comment by TheLarkInn** [comment]
Closed and reopened to kick off a new build.

**Comment by timse** [comment]
checking why this fails!

**Comment by timse** [comment]
found it. The only user of Dependency.compare seams to be Compilation, maybe we can inline that actually?

**Comment by TheLarkInn** [comment]
Well freaking done!!! That was a huge one.

**Comment by TheLarkInn** [comment]
Wow thank you so much. Would you mind shooting me an email? You can find it on my profile page. I'd like to chat with you briefly <3

**Pull request merged**